variables.tf

```diff
@@ -1,4 +1,137 @@
+  type        = bool
+  default     = true
+  description = <<-EOT
+    Enables an authorization rule and route for the VPN to access the internet.
+    Please note, you must allow ingress/egress to the internet (0.0.0.0/0) via the Subnet's security group.
+  EOT
+}
+
+variable "organization_name" {
+  type        = string
+  description = "Name of organization to use in private certificate"
+}
+
+variable "retention_in_days" {
+  description = "Number of days you want to retain log events in the log group"
+  default     = "30"
+}
+
+variable "logging_stream_name" {
+  type        = string
+  description = "Names of stream used for logging"
+}
+
+variable "basic_constraints" {
+  description = <<-EOT
+    The [basic constraints](https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.9) of the issued certificate.
+    Currently, only the `CA` constraint (which identifies whether the subject of the certificate is a CA) can be set.
+    Defaults to this certificate not being a CA.
+  EOT
+  type = object({
+    ca = bool
+  })
+  default = {
+    ca = false
+  }
+}
+
+variable "validity" {
+  description = <<-EOT
+    Validity settings for the issued certificate:
+    `duration_hours`: The number of hours from issuing the certificate until it becomes invalid.
+    `early_renewal_hours`: If set, the resource will consider the certificate to have expired the given number of hours before its actual expiry time (see: [self_signed_cert.early_renewal_hours](https://registry.terraform.io/providers/hashicorp/tls/latest/docs/resources/self_signed_cert#early_renewal_hours)).
+    Defaults to 10 years and no early renewal hours.
+  EOT
+  type = object({
+    duration_hours      = number
+    early_renewal_hours = number
+  })
+  default = {
+    duration_hours      = 87600
+    early_renewal_hours = null
+  }
+}
+
+variable "saml_metadata_document" {
+  default     = null
+  description = "Optional SAML metadata document. Must include this or `saml_provider_arn`"
+  type        = string
+}
+
+variable "saml_provider_arn" {
+  default     = null
+  description = "Optional SAML provider ARN. Must include this or `saml_metadata_document`"
+  type        = string
+
+  validation {
+    error_message = "Invalid SAML provider ARN."
+
+    condition = (
+      var.saml_provider_arn == null ||
+      try(length(regexall(
+        "^arn:aws:iam::(?P<account_id>\\d{12}):saml-provider/(?P<provider_name>[\\w+=,\\.@-]+)$",
+        var.saml_provider_arn
+        )) > 0,
+        false
+    ))
+  }
+}
+
+variable "additional_routes" {
+  default     = []
+  description = "A list of additional routes that should be attached to the Client VPN endpoint"
+
+  type = list(object({
+    destination_cidr_block = string
+    description            = string
+    target_vpc_subnet_id   = string
+  }))
+}
+
+variable "additional_security_groups" {
+  default     = []
+  description = "List of security groups to attach to the client vpn network associations"
```

Comment:
I think we are not using our security group module

Reply:
https://github.com/cloudposse/terraform-aws-security-group

Reply:
https://github.com/cloudposse/terraform-aws-elasticache-redis/blob/master/main.tf#L17-L28